### PR TITLE
testing file hierarchy

### PR DESCRIPTION
**Pull request by jspaaks**
**List of related issues or pull requests**

Refs: #265 #267 #274

**Describe the changes made in this pull request**

1. `repo` is now the first branch in the file hierarchy under `tests`, because that is more meaningful than `platform` (#265). As a result, it is now easier to have tests implemented for the same scenario on all supported platforms. Additionally, describing the cases with human readable annotations and showing those as `describe` and `it` using pytest hooks is now more feasible (#266).
2. `tests/repo0` is now the "null-scenario" and `tests/repo1` is the "everything scenario". Refs #267
3. The local testing data directory `output-files` was renamed to `snippets`. Refs #274 
4. Added directory `frontend-files` that contains files related to the mocked frontend of github, gitlab, etc; in particular `index.html`. Updated helper functions.
5. Updated `repo1/github` to have all `True`s
6. added `repo1/gitlab` tests equivalent to `repo1/github`
7. Made mocked routes minimal for `repo0` cases
8. Moved `skip_unreachable` decorator to `tests/helpers`
9. Replaced specific names for owner and repo with generic ones
10. There are now 268 passing tests, no failing ones; SonarCloud reports a code coverage of ~80.7%~ 78.2% after merge

Here is the new file structure for a given case:

```text
.
├── github
│   ├── conftest.py
│   ├── frontend-files
│   │   └── index.html
│   ├── __init__.py
│   ├── snippets
│   │   └── cli-no-args.txt
│   ├── test_checker_*.py
│   ├── test_cli_*.py
│   └── test_repo_*.py
├── gitlab
│   ├── conftest.py
│   ├── frontend-files
│   │   └── index.html
│   ├── __init__.py
│   ├── snippets
│   │   └── cli-no-args.txt
│   ├── test_checker_*.py
│   ├── test_cli_*.py
│   └── test_repo_*.py
├── __init__.py
├── repo-files
│   ├── CITATION
│   ├── CITATION.cff
│   ├── codemeta.json
│   ├── .howfairis.yml
│   ├── README.rst
│   └── .zenodo.json
└── user-files
    ├── .howfairis-default.yml
    ├── .howfairis-emptyreasons.yml
    └── .howfairis-skipreasons.yml


```

**Instructions to review the pull request**

```
# in an activated virtual env with dev tools installed
pytest
```

### CODE REVIEW

File: tests/helpers/load_repo_files_from_local_data.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from .load_files_from_local_data import load_files_from_local_data
+
+
+def load_repo_files_from_local_data(who_is_asking_file):
+    return load_files_from_local_data(who_is_asking_file, dir_type="repo")
+
+
```
Comment (fdiblen) [inline tests/helpers/load_repo_files_from_local_data.py:7]:
```suggestion
def load_repo_files_from_local_data(who_is_asking_file):
    return load_files_from_local_data(who_is_asking_file, dir_type="repo")
```

File: tests/helpers/list_files_from_local_data.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+def list_files_from_local_data(who_is_asking_file, dir_type):
+    dir_types = ["frontend", "output", "repo", "user"]
+    assert dir_type in dir_types, "Argument dir_type should be one of {0}".format(str(dir_types))
+
+    parent_dir = os.path.dirname(os.path.realpath(who_is_asking_file))
+
+    d = {
+        "frontend": "frontend-files",
+        "output": "snippets",
```
Comment (fdiblen) [inline tests/helpers/list_files_from_local_data.py:12]: Can we call `output` as `snippets` as well for consistency. We should also rename it in other files.

File: tests/repo1/github/conftest.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import pytest
```
Comment (fdiblen) [inline tests/repo1/github/conftest.py:1]: Can we please rename the file as show below?
tests/case1/github/conftest.py

File: tests/repo1/github/test_checker_no_args.py [hunk 4]
```diff
@@ -6,33 +6,33 @@
 
 
 def get_checker():
-    repo = Repo("https://github.com/fair-software/badge")
+    repo = Repo("https://github.com/owner1/repo1")
```
Comment (fdiblen) [inline tests/repo1/github/test_checker_no_args.py:9]:
```suggestion
    repo = Repo("https://github.com/fair-software/case1")
```

**Review by fdiblen — CHANGES_REQUESTED** [review]
Amazing job! :meat_on_bone: 
I made a few suggestions.

**Comment by jspaaks** [comment]
Thanks for the feedback, I'll get right on it

File: tests/helpers/list_files_from_local_data.py (hunk 2 above)
Comment (jspaaks) [reply tests/helpers/list_files_from_local_data.py:12]: done in upcoming commit

File: tests/repo1/github/conftest.py (hunk 3 above)
Comment (jspaaks) [reply tests/repo1/github/conftest.py:1]: done in upcoming commit

File: tests/repo1/github/test_checker_no_args.py (hunk 4 above)
Comment (jspaaks) [reply tests/repo1/github/test_checker_no_args.py:9]: done here and everywhere else

**Comment by jspaaks** [comment]
Hi Faruk, I made the changes, can you confirm this is now good to merge? Thanks!

**Pull request merged**